### PR TITLE
set window title to "Warcraft Recorder"

### LINKED ISSUE #55: Window title (Taskbar/alt+tab etc) is "Arena Recorder"

### PR DESCRIPTION
**Pull request by csutcliff**
fixes #55

### CODE REVIEW

**Comment by ericlytle** [comment]
PR title is a little misleading as this also seems to be attempting to add M+ support

File: src/main/logutils.ts [hunk 1]
```diff
@@ -117,9 +122,13 @@ const handleLogLine = (line: string) => {
...
     } else if (line.includes("ENCOUNTER_START")) {
-        handleRaidStartLine(line);
+        handleEncounterStartLine(line);
     } else if (line.includes("ENCOUNTER_END")) {
-        handleRaidStopLine(line);
+        handleEncounterStopLine(line);
+    } else if (line.includes("CHALLENGE_MODE_START")) {
+        handleChallengeModeStartLine(line);
+    } else if (line.includes("CHALLENGE_MODE_END")) {
+        handleChallengeModeStopLine(line);
```
Comment (ericlytle) [inline src/main/logutils.ts:131]: if a group disbands, i dont think we get the challenge mode end event, right?

File: src/main/logutils.ts [hunk 2]
```diff
@@ -262,6 +286,69 @@ const determineArenaMatchResult = (line: string): any[] => {
...
+     * so in theory if you start a new keystone before a failed key has been stopped the
+     * video will still be stopped/started in quick succession to ensure a separation.
+     */
+
+    const encounterID = parseInt(line.split(',')[1]);
+    const category = "ChallengeModes";
+
+    videoStartDate = getCombatLogDate(line);
+
+    metadata = {
```
Comment (ericlytle) [inline src/main/logutils.ts:304]: we probably want to capture the key level too

this is actually why i stopped working on this. I originally was just going to add all this detail to this metadata but then i realized we are going to have alot of dead properties depending on the gameplay type. so the best way was to create an abstract class with the common attributes and then have other gameplay types extend that depending on their use case.

**Comment by csutcliff** [comment]
> PR title is a little misleading as this also seems to be attempting to add M+ support

ah that's my fault, I didn't make this a seperate branch so when I continued working on the m+ support the PR here reflected that. Still fairly new to Git. I'll see what I can do to sort it out.

File: src/main/logutils.ts (hunk 1 above)
Comment (csutcliff) [reply src/main/logutils.ts:131]: No, I was planning on working on a combat log timeout for this before I sent it in a PR. Reading the comments I believe that could help for BGs too?

File: src/main/logutils.ts (hunk 2 above)
Comment (csutcliff) [reply src/main/logutils.ts:304]: That makes sense, the key level is certainly something useful to display to the user.

**Pull request closed without merging**